src/pages/courseDetailsPage/CourseDetailsPage.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from "react";
+import { useLocation, useParams } from "react-router-dom";
+import { useGetCourseByIdQuery } from "redux/coursesApi";
+import {
+  ArrowIcon,
+  BackLink,
+  CourseDescription,
+  CourseLesson,
+  CourseLessonsList,
+  LessonTitle,
+  CourseTitle,
+  LockIcon,
+  Title,
+  TitleBox,
+  VideoContainer,
+  CurrentLesson,
+  CurrentLessContainer,
+  CurrentText,
+} from "./CourseDetailsPage.styled";
+import { Container } from "components/coursesList/CoursesList.styled";
+import VideoPlayer from "components/VideoPlayer/VideoPlayer";
+
+interface LessonseDetails {
+  id: string;
+  link: string;
+  title: string;
+  status: string;
+  previewImageLink: string;
+  order: number;
+}
+
+const CourseDetailsPage: React.FC = () => {
+  const { id: courseId = "" } = useParams();
+  const { state } = useLocation();
+  const backLinkHref = state?.from ?? "/";
+  const { data: course } = useGetCourseByIdQuery(courseId);
+  const [currentLesson, setCurrentLesson] = useState<LessonseDetails | null>(
+    null
+  );
+  const [isVideoOpen, setIsVideoOpen] = useState(true);
+
+  useEffect(() => {
+    if (course && course.lessons && course.lessons.length > 0) {
+      setCurrentLesson(course.lessons[0]);
+    }
+  }, [course]);
+
+  const handleLessonClick = (lesson: LessonseDetails) => {
+    if (lesson.status !== "locked") {
+      setIsVideoOpen(true);
+      setCurrentLesson(lesson);
+    }
+    if (lesson.status === "locked") {
+      setIsVideoOpen(false);
+    }
+  };
+
+  return (
+    <>
```

Comment:
Everything is wrapped in a container. He is unnecessary here

Reply:
Done